### PR TITLE
[ADD] a backup dashboard section with buttons: "Update Info", "Up to …

### PR DESCRIPTION
**Pull request by ufaks**
…the balance", "Make Backup"

### CODE REVIEW

File: odoo_backup_sh/controllers/main.py [hunk 1]
```diff
@@ -48,26 +55,31 @@ def _render_template(self, **d):
 
 class BackupController(http.Controller):
 
-    @http.route('/web/database/backups', type='http', auth="none")
-    def backup_list(self):
-        config._parse_config()
-        user_key = config.get('odoo_backup_user_key')
+    def backup_service_request(self, redirect=None):
```
Comment (yelizariev) [inline odoo_backup_sh/controllers/main.py]: ``load_backup_list``?

File: odoo_backup_sh/models/odoo_backup_sh.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+from odoo.addons.iap import jsonrpc, InsufficientCreditError
+
+
+class Dashboard(models.Model):
+    _name = 'odoo_backup_sh.dashboard'
+    _description = 'Backup Dashboard'
+
+    @api.model
+    def action_dashboard_redirect(self):
+        return self.env.ref('odoo_backup_sh.backup_dashboard').read()[0]
```
Comment (yelizariev) [inline odoo_backup_sh/models/odoo_backup_sh.py]: what is ``.read()[0]``? Can you just use ``.id``?

Do you need this method besides ``ir.actions.server``? If not, you could this move there 
``action = env.ref('odoo_backup_sh.backup_dashboard')...``

File: odoo_backup_sh/models/odoo_backup_sh.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import werkzeug
+
+import odoo
+from odoo import http, api, fields, models
+from odoo.http import request
+from odoo.addons.odoo_backup_sh.controllers.main import BackupController, BACKUP_SERVICE_ENDPOINT
+from odoo.addons.iap import jsonrpc, InsufficientCreditError
+
+
+class Dashboard(models.Model):
```
Comment (yelizariev) [inline odoo_backup_sh/models/odoo_backup_sh.py]: Are you sure that it cannot be ``models.TransientModel`` or even ``models.AbstractModel``?

File: odoo_backup_sh/models/odoo_backup_sh.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    date_upload = fields.Datetime(string='Upload Date')
+
+    @api.model
+    def update_info(self, redirect):
+        response = BackupController().backup_service_request(redirect=redirect)
+        if 'auth_link' not in response:
+            credit_url = self.check_insufficient_credit(credit=10)
+            if credit_url:
+                response['credit_url'] = credit_url
+        return response
```
Comment (yelizariev) [inline odoo_backup_sh/models/odoo_backup_sh.py]: I guess you need to add TODO to save new backup list?

File: odoo_backup_sh/models/odoo_backup_sh.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+    @api.model
+    def make_backup(self, name=None, backup_format='zip'):
+        if name is None:
+            name = self.env.cr.dbname
+        # ts = datetime.datetime.utcnow().strftime("%Y-%m-%d_%H-%M-%S")
+        # filename = "%s_%s.%s" % (name, ts, backup_format)
+        # dump_stream = odoo.service.db.dump_db(name, None, backup_format)
+        # return dump_stream
```
Comment (yelizariev) [inline odoo_backup_sh/models/odoo_backup_sh.py]: Put todo with some details, probably. For example 

```
TODO make dump
TODO get url to upload
TODO upload

```

File: odoo_backup_sh/controllers/main.py (hunk 1 above)
Comment (ufaks) [reply odoo_backup_sh/controllers/main.py]: ok, but you have taken into account that the function will return auth_link if the user is not logged on odoo-backup.sh

File: odoo_backup_sh/models/odoo_backup_sh.py (hunk 2 above)
Comment (ufaks) [reply odoo_backup_sh/models/odoo_backup_sh.py]: .read()[0] returns record dict mapping field names to their values, like this
{'create_uid': (1, 'Administrator'), 'xml_id': 'odoo_backup_sh.backup_dashboard', 'context': '{}' ...}
with .id odoo raises attribute error

- "Do you need this method besides ir.actions.server?"
I think it's a good way to manage action views in the future

File: odoo_backup_sh/controllers/main.py (hunk 1 above)
Comment (yelizariev) [reply odoo_backup_sh/controllers/main.py]: ``load_backup_list_from_service`` + *note in method description about auth link*?

File: odoo_backup_sh/models/odoo_backup_sh.py (hunk 2 above)
Comment (yelizariev) [reply odoo_backup_sh/models/odoo_backup_sh.py]: Ok, odoo uses the same approach

**Comment by yelizariev** [comment]
Still lint warnings:

odoo_backup_sh/controllers/main.py:16:1: F401 'werkzeug.utils' imported but unused
odoo_backup_sh/models/odoo_backup_sh.py:6:1: F401 'werkzeug' imported but unused
odoo_backup_sh/models/odoo_backup_sh.py:8:1: F401 'odoo' imported but unused
odoo_backup_sh/models/odoo_backup_sh.py:9:1: F401 'odoo.http' imported but unused
odoo_backup_sh/models/odoo_backup_sh.py:10:1: F401 'odoo.http.request' imported but unused

File: odoo_backup_sh/tests/test_odoo_backup_sh_ui.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+# Written by Stanislav Krotov <https://www.it-projects.info/team/ufaks>, April 2018
+
+import odoo.tests
+try:
+    from unittest.mock import patch
+except ImportError:
+    from mock import patch
+
+
+@odoo.tests.common.at_install(False)
```
Comment (yelizariev) [inline odoo_backup_sh/tests/test_odoo_backup_sh_ui.py]: You must explain why it has at_install False

File: odoo_backup_sh/tests/test_odoo_backup_sh_ui.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        def odoo_backup_sh_response(self, redirect=None):
+            return {'credit_url': 'http://odoo.com'}
+
+        self.patcher = patch(
+            'odoo.addons.odoo_backup_sh.controllers.main.BackupController.load_backup_list_from_service',
+            wraps=odoo_backup_sh_response)
+        self.patcher.start()
+
+    def test_01_odoo_backup_sh_tour(self):
+        self.phantom_js('/', "odoo.__DEBUG__.services['web_tour.tour'].run('odoo_backup_sh_tour')",
```
Comment (yelizariev) [inline odoo_backup_sh/tests/test_odoo_backup_sh_ui.py]: Why ``/``? It's backend tour

File: odoo_backup_sh/views/odoo_backup_sh_templates.xml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    <template id="assets_backend" inherit_id="web.assets_backend" name="odoo_backup_sh assets">
+        <xpath expr="." position="inside">
+            <link rel="stylesheet" href="/odoo_backup_sh/static/src/less/dashboard.less"/>
+        </xpath>
+        <xpath expr="//script[last()]" position="after">
+            <script type="text/javascript" src="/odoo_backup_sh/static/src/js/dashboard.js"/>
+        </xpath>
+    </template>
+
+    <template id="assets_common" name="tour" inherit_id="web.assets_common">
```
Comment (yelizariev) [inline odoo_backup_sh/views/odoo_backup_sh_templates.xml]: use ``assets_backend``. Put the tour in demo file

File: odoo_backup_sh/__manifest__.py [hunk 9]
```diff
@@ -21,9 +21,15 @@
...
-    "qweb": [],
-    "demo": [],
+    "data": [
+        'security/ir.model.access.csv',
+        'views/odoo_backup_sh_templates.xml',
+        'views/odoo_backup_sh_views.xml',
+    ],
+    'qweb': ['static/src/xml/dashboard.xml'],
+    "demo": [
+        'demo/odoo_backup_sh_demo.xml',
```
Comment (yelizariev) [inline odoo_backup_sh/__manifest__.py]: rename to ``tour_views.xml`` please

File: odoo_backup_sh/tests/test_odoo_backup_sh_ui.py [hunk 10]
```diff
@@ -33,6 +34,15 @@ def patch_check_insufficient_credit(credit):
...
     def test_01_odoo_backup_sh_tour(self):
+        # needed because tests are run before the module is marked as
+        # installed. In js web will only load qweb coming from modules
+        # that are returned by the backend in module_boot. Without
+        # this you end up with js, css but no qweb.
+        cr = self.registry.cursor()
+        assert cr == self.registry.test_cr
+        env = Environment(cr, self.uid, {})
+        env['ir.module.module'].search([('name', '=', 'odoo_backup_sh')], limit=1).state = 'installed'
+        cr.release()
```
Comment (yelizariev) [inline odoo_backup_sh/tests/test_odoo_backup_sh_ui.py]: You don't need to release if you don't use acquire

**Pull request merged**